syntax = "proto2";

package proto;

message Connection {
    required string server_ip = 1;
    required uint32 server_port = 2;
    required string client_ip = 3;
    required uint32 client_port = 4;

    optional string client_service_name = 5;
    optional string server_service_name = 6;
}

message RequestLog {
    required string trace_id = 1;
    required string span_id = 2;

    // The number of seconds since epoch this request was sent/received
    required int64 time = 3;

    required double duration = 4;

    required Connection conn = 5;

    // The number of transactions that were sent through conn,
    // including this one
    required uint32 transaction_count = 6;

    // Indicates the role of this machine in this connection
    enum Role {
        CLIENT = 0;
        SERVER = 1;
    }
    required Role role = 7;
}
